lib/addons/aws-node-termination-handler/index.ts

```diff
@@ -66,10 +66,14 @@ export class AwsNodeTerminationHandlerAddOn extends HelmAddOn {
    */
   deploy(clusterInfo: ClusterInfo): void {
     const cluster = clusterInfo.cluster;    
-    const asgCapacity = clusterInfo.autoscalingGroups;
+    const asgCapacity = clusterInfo.autoscalingGroups || [];
 
-    // No support for Fargate and Managed Node Groups, lets catch that
-    assert(asgCapacity && asgCapacity.length > 0, 'AWS Node Termination Handler is only supported for self-managed nodes');
+    const karpenter = clusterInfo.getScheduledAddOn('KarpenterAddOn');
+    if (!karpenter){
```

Comment:
What is your handling for ASG tagging applied in queue mode here: https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/aws-node-termination-handler/index.ts#L151

Reply:
This is handled in the latest commit - please take a look when you can.